### ISSUE TITLE
Getting PIN email, but it doesn't work

### ISSUE THREAD

**Issue by mflagler** [ReneNulschDE/mbapi2020#192]
Seeing unknown error occurred, but also getting these errors in the logs (guessing the first one is the one that triggers the prompt to reconfigure after updating to the latest version that was just pushed):
Home Assistant v2024.1.6
Integration v0.10.0

```
Logger: homeassistant.config_entries
Source: config_entries.py:430
First occurred: 8:21:21 PM (1 occurrences)
Last logged: 8:21:21 PM

Config entry 'GLE350' for mbapi2020 integration could not authenticate: 400, message='Bad Request', url=URL('https://id.mercedes-benz.com/as/token.oauth2')
```

```
This error originated from a custom integration.

Logger: custom_components.mbapi2020.oauth
Source: custom_components/mbapi2020/oauth.py:274
Integration: MercedesME 2020 (documentation, issues)
First occurred: 8:21:21 PM (2 occurrences)
Last logged: 8:25:12 PM

ClientError requesting data from https://id.mercedes-benz.com/as/token.oauth2: 400, message='Bad Request', url=URL('https://id.mercedes-benz.com/as/token.oauth2')
```

**Comment by ReneNulschDE** [comment]
Does this happen when you add the integration or in the normal operation?. Could you share your region? Is this the first/only mb account or do you use multiple accounts?

Could you please share the system details too? (Can be found in: [Settings ⇒ System ⇒ Repairs ⇒ Three Dots in Upper Right ⇒ System information](https://my.home-assistant.io/redirect/system_health/))

thx

**Comment by Whatsek** [comment]
Same problem, after upgrading to Integration v0.10.0, deleting the token and reconfiguring the plugin with email adres and the received code, my car is detected but all entities are unknown. (Code is accepted and the plugin loads fine)

This is in my system info:

MercedesME 2020
MB API reachable	ok
MB WS state	unknown
Connected cars	0
Version	v0.10.0

Region is Europe, one account, worked fine for a long time.

**Comment by ReneNulschDE** [comment]
I can reproduce it. Working on a fix.

**Comment by ReneNulschDE** [comment]
Question to all of you: Do you have deleted the integration before upgrading the MBAPI2020 version or did you an direct update and restart?

Could you please check the following workaround:
1. Delete the token file in your "HA config dir"/.storage/.mercedesme-token-cache-....
2. Restart HA
3. Add the integration again

**Comment by Pieter-HA** [comment]
I did a direct update and restart of HA because the PIN didn't come through email. Upgrade to 0.10.0 made the PIN come through in email, but raised an error on entry in teh add-on. Same for the beta version.
Deleted the token cache after every update / restart.

Don't understand your point 3. Why add the integration again if it wasn't deleted but just updated.

**Comment by Pieter-HA** [comment]
@ReneNulschDE: I deleted the integration, deleted the mbapi2020 directory in custom_components and restarted HA.
Re-installed the integration from HACS (v.0.10.0), restarted HA (v.2024.2.0). 
Added the integration and SUCCESS!
Thanks for your help.

**Comment by mflagler** [comment]
Following the steps also fixed it for me on HA v2024.1.6

**Comment by ReneNulschDE** [comment]
Solved with v0.10.1
